ambry-server/src/main/java/com/github/ambry/server/RepairRequestsSender.java

```diff
@@ -0,0 +1,242 @@
+ */
+class RepairRequestsSender implements Runnable {
+  private static final Logger logger = LoggerFactory.getLogger(RepairRequestsSender.class);
+  private static final String SERVICE_ID = "RepairRequestsSender";
+  // ODR can take some time to execute due to remote call to the peer replica.
+  private static int POLL_TIMEOUT_MS = 10000;
+  private static int SLEEP_TIME_MS = 5000;
+
+  // the channel to talk to the requests handler
+  private final RequestResponseChannel requestChannel;
+  // network client to send the requests and poll the responses
+  private final LocalNetworkClient client;
+  // cluster map
+  private final ClusterMap clusterMap;
+  // this data node
+  private final DataNodeId nodeId;
+  // LOCAL_CONSISTENCY_TODO. May listen to the Helix to get updated partition lists.
+  private final ClusterParticipant clusterParticipant;
+  // The database which stores the RepairRequestRecord
+  private final RepairRequestsDb db;
+  // partition id list this data node gets assigned
+  private final List<Long> partitionIds;
+  // Hashmap from partition id to the replicas
+  private final Map<Long, ReplicaId> partition2Replicas;
+  // generator to generate the correlation ids.
+  private final AtomicInteger correlationIdGenerator = new AtomicInteger(0);
+  // max result sets it can get from each query
+  private final int maxResults;
+  // shutdown flag
+  private volatile boolean shutdown = false;
+
+  /**
+   * RepairRequestsSender constructor
+   * @param requestChannel the channel to send requests and receive responses
+   * @param factory the {@link LocalNetworkClientFactory} used to generate the {@link LocalNetworkClient}
+   * @param clusterMap the cluster map
+   * @param nodeId this {@link DataNodeId}
+   * @param repairRequestsDb the database to get the RepairRequestRecord
+   * @param clusterParticipant {@link ClusterParticipant}
+   */
+  public RepairRequestsSender(RequestResponseChannel requestChannel, LocalNetworkClientFactory factory,
+      ClusterMap clusterMap, DataNodeId nodeId, RepairRequestsDb repairRequestsDb,
+      ClusterParticipant clusterParticipant) {
+    this.requestChannel = requestChannel;
+    this.client = factory.getNetworkClient();
+    this.clusterMap = clusterMap;
+    this.nodeId = nodeId;
+    this.clusterParticipant = clusterParticipant;
+
+    this.db = repairRequestsDb;
+    this.maxResults = db.getListMaxResults();
+
+    // LOCAL_CONSISTENCY_TODO. listen to the clustermap change and make the modification accordingly.
+    partitionIds = new ArrayList<>();
+    partition2Replicas = new HashMap<>();
+    for (ReplicaId replica : clusterMap.getReplicaIds(nodeId)) {
+      long partition = replica.getPartitionId().getId();
+      partitionIds.add(partition);
+      partition2Replicas.put(partition, replica);
+    }
+  }
+
+  public void run() {
+    while (!shutdown) {
+      try {
+        // sleep for some time
+        Thread.sleep(SLEEP_TIME_MS);
```

Comment:
can we put the sleep in the end of this loop?

Reply:
Will change it in the metrics PR.